letsencrypt/models/letsencrypt.py

```diff
@@ -0,0 +1,157 @@
+                .get_param('letsencrypt.altname.%d' % i)
+            )
+            i += 1
+        _logger.info('generating csr for %s', domain)
+        if len(domains) > 1:
+            _logger.info('with alternative subjects %s', ','.join(domains[1:]))
+        config = self.env['ir.config_parameter'].get_param(
+            'letsencrypt.openssl.cnf', '/etc/ssl/openssl.cnf')
+        csr = os.path.join(self.get_data_dir(), '%s.csr' % domain)
+        with tempfile.NamedTemporaryFile() as cfg:
+            cfg.write(open(config).read())
+            if len(domains) > 1:
+                cfg.write(
+                    '\n[SAN]\nsubjectAltName=' +
+                    ','.join(map(lambda x: 'DNS:%s' % x, domains)) + '\n')
+            cfg.file.flush()
+            cmdline = [
+                'openssl', 'req', '-new',
+                self.env['ir.config_parameter'].get_param(
+                    'letsencrypt.openssl.digest', '-sha256'),
+                '-key', self.generate_domain_key(domain),
+                '-subj', '/CN=%s' % domain, '-config', cfg.name,
+                '-out', csr,
+            ]
+            if len(domains) > 1:
+                cmdline.extend([
+                    '-reqexts', 'SAN',
+                ])
+            self.call_cmdline(cmdline)
+        return csr
+
+    @api.model
+    def cron(self):
+        domain = urlparse.urlparse(
+            self.env['ir.config_parameter'].get_param(
+                'web.base.url', 'localhost')).netloc
+        self.validate_domain(domain)
+        account_key = self.generate_account_key()
+        csr = self.generate_csr(domain)
+        manifest, manifest_path = file_open(
+            'letsencrypt/__openerp__.py', pathinfo=True)
+        manifest.close()
+        acme_challenge = os.path.join(
+            os.path.dirname(manifest_path),
+            'static', 'acme-challenge')
+        if not os.path.isdir(acme_challenge):
+            os.makedirs(acme_challenge)
+        if self.env.context.get('letsencrypt_dry_run'):
+            crt_text = 'I\'m a test text'
+        else:  # pragma: no cover
+            from .acme_tiny import get_crt, DEFAULT_CA
+            crt_text = get_crt(
+                account_key, csr, acme_challenge, log=_logger, CA=DEFAULT_CA)
+        with open(os.path.join(self.get_data_dir(), '%s.crt' % domain), 'w')\
+                as crt:
+            crt.write(crt_text)
+            chain_cert = urllib2.urlopen(
+                self.env['ir.config_parameter'].get_param(
+                    'letsencrypt.chain_certificate_address',
+                    'https://letsencrypt.org/certs/'
+                    'lets-encrypt-x1-cross-signed.pem')
+            )
+            crt.write(chain_cert.read())
+            chain_cert.close()
+            _logger.info('wrote %s', crt.name)
+        self.call_cmdline([
+            'sh', '-c', self.env['ir.config_parameter'].get_param(
+                'letsencrypt.reload_command', ''),
```

Comment:
Opening an empty subprocess when you can simply skip it with an `if` or `try` makes little sense IMHO. Please change that.